bitnami/mariadb/templates/secrets.yaml

```
@@ -13,25 +13,23 @@ metadata:
   {{- end }}
 type: Opaque
 data:
-  {{- if not (empty .Values.auth.rootPassword) }}
-  mariadb-root-password: {{ .Values.auth.rootPassword | b64enc | quote }}
-  {{- else if (not .Values.auth.forcePassword) }}
-  mariadb-root-password: {{ randAlphaNum 10 | b64enc | quote }}
+  {{- if (not .Values.auth.forcePassword) }}
+  mariadb-root-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mariadb-root-password" "providedValues" (list "auth.rootPassword") "context" $) }}
   {{- else }}
   mariadb-root-password: {{ required "A MariaDB Root Password is required!" .Values.auth.rootPassword }}
   {{- end }}
-  {{- if and (not (empty .Values.auth.username)) (not (empty .Values.auth.password)) }}
-  mariadb-password: {{ .Values.auth.password | b64enc | quote }}
-  {{- else if (not .Values.auth.forcePassword) }}
-  mariadb-password: {{ randAlphaNum 10 | b64enc | quote }}
+  {{- if (not .Values.auth.forcePassword) }}
+  {{- $passwordValuePath := "" }}
+  {{- if not (empty .Values.auth.username) }}
+  {{- $passwordValuePath := "auth.password" }}
+  {{- end }}
+  mariadb-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mariadb-password" "providedValues" (list $passwordValuePath) "context" $) }}
   {{- else }}
   mariadb-password: {{ required "A MariaDB Database Password is required!" .Values.auth.password }}
   {{- end }}
```

I think we should reconsider this logic. If `auth.username` is not provided, this key will never be used so there is no point in generating it or forcing it to be filled if `auth.forcePassword=true`.

```suggestion
  {{- if (not (empty .Values.auth.username)) }}
  {{- if (not .Values.auth.forcePassword) }}
  mariadb-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "mariadb-password" "providedValues" (list "auth.password") "context" $) }}
  {{- else }}
  mariadb-password: {{ required "A MariaDB Database Password is required!" .Values.auth.password | b64enc | quote }}
  {{- end }}
  {{- end }}
```
```

---

Great catch! 👍

Just for me to understand: The current implementation will cause the helm installation to fail with "A MariaDB Database Password is required!" if `auth.username` is empty and `auth.forcePassword` is true. This won't be the case anymore with your suggestion. Is that meant to be?